### PR TITLE
Add old-channels command

### PR DESCRIPTION
**Pull request by williambl**
Locked to console channel.

### CODE REVIEW

**Review by jriwanek — APPROVED** [review]

**Comment by KiriCattus** [comment]
When tested we found out that this command tries to loop over the channels but just keeps on going and adding ones it has already gotten.

**Comment by williambl** [comment]
It seems like it works now. Perhaps updating JDA fixed it.

**Comment by williambl** [comment]
No, that can't be it - JDA isn't updated in this branch.

Whatever the reason, it seems to work.

**Comment by KiriCattus** [comment]
I'll give it a test again on the larger server in a little bit, last I remember it just kept on looping through the server and getting duplication's of multiple channels

**Comment by KiriCattus** [comment]
I've tested out the channels command again and threw in a few breakpoints on lines 46, 62 and 78 with and without lines 41 to 44 disabled, nothing happened in discord after the command ``!mmd-old-channels`` was run and nothing showed up in the logs or caught on the breakpoints.

**Comment by williambl** [comment]
Okay, this should be working fine now, and merge conflicts are fixed. Should be ready for merge after review.

image

File: src/main/java/com/mcmoddev/mmdbot/updatenotifiers/forge/ForgeVersionHelper.java [hunk 1]
```diff
@@ -43,7 +44,7 @@ public static MinecraftForgeVersion getLatestMcVersionForgeVersions() throws Exc
         return new MinecraftForgeVersion(latest, versions.get(latest));
     }
 
-    private static InputStreamReader openUrl() throws Exception {
+    private static InputStreamReader openUrl() throws IOException {
```
Comment (williambl) [inline src/main/java/com/mcmoddev/mmdbot/updatenotifiers/forge/ForgeVersionHelper.java]: Didn't realise I'd committed this :|

**Comment by KiriCattus** [comment]
Suggestion to try with this, I noted with the relay bot and the mod we where using that the bot would continue to loop over all the channels in the entire server until I blocked the relay bot from being able to see the community channels (reducing the number of channels it could see therefore somehow stopping the bot looping and breaking) maybe we should limit this command to the community mods category since that is the main if not the only place it should be checking?

**Comment by williambl** [comment]
I thought last we checked this worked?

**Comment by KiriCattus** [comment]
I thought the last time I ran it locally and tested it that we still had channel looping but I can go and double check it now to be sure because I could be wrong, if I am I'll take that back and apologise for sure

**Review by jriwanek — APPROVED** [review]

**Comment by KiriCattus** [comment]
Update: we found the issues, listed them in discord last night, channel perm checks need to be in place or the timer fails entirely, if a perm is missing it should skip that channel as if it does not exist, category checking is broken (Or so Will says) channels can't be mentioned, only typed out for example ``proxys-trinkets`` not ``#proxys-trinkets`` but other than that things are all good :) my assumption was incorrect and I was remembering a different now resolved issue so my apologies :s

**Review by jriwanek — APPROVED** [review]

**Comment by KiriCattus** [comment]
If we can figure out that metals channel issue where it does not say when the last message was (currently it says 0 when the last message was a while ago) and add in the following we should be good for this PR :D
Missing channel perms catch/exception for ${channels}
Non-existing channel/category reply ${channel/category}
(Paste from Discord chat so we don't loose track like I did last time...)

**Pull request merged**